### PR TITLE
add entityNameExp to Group for device.name definition

### PR DESCRIPTION
**Pull request by AlvaroVega**
https://github.com/telefonicaid/iotagent-node-lib/issues/1222
https://github.com/telefonicaid/iotagent-node-lib/issues/1145

### CODE REVIEW

**Comment by AlvaroVega** [comment]
Another PRs related and needed: https://github.com/telefonicaid/iotagent-manager/pull/290 https://github.com/telefonicasc/iotp-portal/pull/216

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -1,3 +1,4 @@
+- Add entityNameExp to Group Model to allow define device.name using expressions
```
[the remark this answers is not in the document]
Comment (AlvaroVega) [reply CHANGES_NEXT_RELEASE]: fixed in https://github.com/telefonicaid/iotagent-node-lib/pull/1255/commits/d0d93758fd9c2ac60ac52c034bba1f21799968f4

File: test/unit/ngsiv2/provisioning/device-provisioning-api_test.js [hunk 2]
```diff
@@ -5,7 +5,7 @@
  *
  * fiware-iotagent-lib is free software: you can redistribute it and/or
  * modify it under the terms of the GNU Affero General Public License as
- * published by the Free Software Foundation, either version 3 of the License,
+ * published by the ree Software Foundation, either version 3 of the License,
```
Comment (fgalan) [inline test/unit/ngsiv2/provisioning/device-provisioning-api_test.js:8]:
```suggestion
 * published by the Free Software Foundation, either version 3 of the License,
```

File: CHANGES_NEXT_RELEASE [hunk 3]
```diff
@@ -1,3 +1,4 @@
+- Add entityNameExp to Group Model to allow define device.name using expressions (#1222, #1145)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:1]:
```suggestion
- Add: entityNameExp to Group Model to allow define device.name using expressions (#1222, #1145)
```

File: doc/api.md [hunk 4]
```diff
@@ -76,7 +76,7 @@ The table below shows the information held in the service group provisioning res
 correspondence between the API resource fields and the same fields in the database model.
 
 | Payload Field                  | DB Field                       | Definition                                                                                                                                                                                                                                                                |
-| ------------------------------ | ------------------------------ | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| ------------------------------ | ------------------------------ | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ---- |
```
Comment (fgalan) [inline doc/api.md:79]:
```suggestion
| ------------------------------ | ------------------------------ | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
```

**Review by fgalan — APPROVED** [review]
LGTM

**Comment by chicco785** [comment]
@fgalan @AlvaroVega not sure 100% but this change require changes in the iot agent manager as well (that should proxy the new configuration parameter)

**Comment by fgalan** [comment]
> @fgalan @AlvaroVega not sure 100% but this change require changes in the iot agent manager as well (that should proxy the new configuration parameter)

PR https://github.com/telefonicaid/iotagent-manager/pull/290 in iotagent-manager is associated to this one. What do think about it? It address your concern or something else is pending in the iotagent-manager?

**Comment by chicco785** [comment]
totally good :) i missed it

**Pull request merged**